### PR TITLE
Ei 364 get employer vendor

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

**Review by David-B-Read — DISMISSED** [review]

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs [hunk 1]
```diff
@@ -23,5 +24,13 @@ public async Task<IActionResult> RefreshVendorRegistrationFormStatus(DateTime fr
 
             return NoContent();
         }
+
+        [HttpPut("legalentities/{hashedLegalEntityId}/employervendorid")]
+        public async Task<IActionResult> GetAndAddEmployerVendorId(string hashedLegalEntityId)
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs:29]: GetAndAdd?  But it doesn't get anything?

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "Vendor": {
+    "VendorIdentifier": "P0004320",
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json:3]: Any reason why most of these these fields begin with Vendor?  They are properties of the Vendor so don't need the prefix.

**Review by Stephek — CHANGES_REQUESTED** [review]
Juts a couple of minor points.

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json:3]: That's a sample response from the Finance API that we are calling.

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs:29]: It's getting the VendorId from BC (Vrf) and adding it the LegalEntity in Employer Incentives.

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/Data/FinanceAPI_V1__VendorDataResponse.json:3]: Yes this is what comes back from BC

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs:29]: Again I can call it AddEmployerVendorIdForLegalEntity("XXX")

Comment (Pauljgraham) [reply src/SFA.DAS.EmployerIncentives.Api/Controllers/LegalEntityController.cs:29]: changed name to AddEmployerVendorIdToLegalEntity()

**Review by Stephek — DISMISSED** [review]

**Review by ruslanss — APPROVED** [review]

**Review by ruslanss — APPROVED** [review]

**Pull request merged**